### PR TITLE
Refactor kustomize config/ folder

### LINKED ISSUE #1572: v1alpha3 - separate webhooks
Separate webhooks as per https://master.cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html

### PR DESCRIPTION
**Pull request by h0tbird**
**What this PR does / why we need it**:
Refactors kustomize `config/` to support multi tenancy as described in https://master.cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html#refactor-kustomize-config-folder-to-support-multi-tenancy-when-using-webhooks

**Which issue(s) this PR fixes**:
Fixes #1572

I am following the documentation and I am also using CAPI as a guide. But something is missing and I am not sure what it is yet:
```
 cluster-api-provider-aws git:(refactor-kustomize) kustomize build config
Error: failed to apply json patch '[{"op":"replace","path":"/spec/conversion/webhook/clientConfig/service/namespace","value":"capi-webhook-system"}]': replace operation does not apply: doc is missing path: /spec/conversion/webhook/clientConfig/service/namespace: missing value
```

### CODE REVIEW

File: config/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+bases:
+- crd
+- webhook # Disable this if you're not using the webhook functionality.
+- default
+
+patchesJson6902:
+- target:
+    group: apiextensions.k8s.io
+    version: v1beta1
```
Comment (h0tbird) [inline config/kustomization.yaml]: I think this should be updated to `v1`

**Comment by vincepri** [comment]
We’ll need to update the CRD to v1 as a pre-requisite

**Comment by h0tbird** [comment]
I have created a PR to update CRDs to `v1`: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1577 then I think this one is ready for review.

**Comment by randomvariable** [comment]
`kustomize build config` is failing on this branch.

/hold

Let me know when this is ready

File: main.go [hunk 2]
```diff
@@ -123,8 +123,8 @@ func main() {
 
 	flag.IntVar(&webhookPort,
 		"webhook-port",
-		9443,
-		"Webhook server port (set to 0 to disable)",
+		0,
+		"Webhook Server port, disabled by default. When enabled, the manager will only work as webhook server, no reconcilers are installed.",
 	)
```
Comment (randomvariable) [inline main.go]: We need some logic in `main.go` to turn off the controllers when the webhook server is enabled, as per https://master.cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html

> The controller MUST register reconcilers if and only if webhookPort == 0.
> The controller MUST register webhooks if and only if webhookPort != 0.

File: config/webhook/kustomization.yaml [hunk 3]
```diff
@@ -1,6 +1,42 @@
+namespace: capi-webhook-system
+
```
Comment (randomvariable) [inline config/webhook/kustomization.yaml]: Need to add namePrefix something to the resources for the webhook as they will live in a shared namespace with other providers.

i.e. service with name `webhook-service` doesn't identify it as the CAPA service.

File: config/webhook/kustomization.yaml [hunk 4]
```diff
@@ -1,6 +1,42 @@
...
 resources:
 - manifests.yaml
 - service.yaml
+- ../certmanager
+- ../manager
```
Comment (randomvariable) [inline config/webhook/kustomization.yaml]: The manager directory contains the secret with AWS credentials. This and associated logic needs to be moved to `config/default`, such that the deployment in `capi-webhook-system` doesn't get AWS credentials.

File: main.go (hunk 2 above)
Comment (h0tbird) [reply main.go]: Thank you! Logic added here https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1575/commits/6f3090142dd512f07fd315d1de508f3ffbc4eef9.

File: config/webhook/kustomization.yaml (hunk 3 above)
Comment (h0tbird) [reply config/webhook/kustomization.yaml]: Already merged in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1577. I have merged master into this PR to get all the required bits including the `namePrefix`.

File: main.go (hunk 2 above)
Comment (randomvariable) [reply main.go]: Great!

File: config/webhook/kustomization.yaml (hunk 4 above)
Comment (randomvariable) [reply config/webhook/kustomization.yaml]: Resolved!

Comment (randomvariable) [reply config/webhook/kustomization.yaml]: Just checked. Two secrets present in both namespaces.

Comment (h0tbird) [reply config/webhook/kustomization.yaml]: @randomvariable let me know if I did the right thing here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1575/commits/1219e68e0d4f1657010397a6aa1760330d7c5745

**Comment by h0tbird** [comment]
> Let me know when this is ready

Tests passed 👏@randomvariable this is ready for review.

**Comment by randomvariable** [comment]
Cool, two final things:
Update the paths in the Makefile on lines 227 & 233
And squash commits, and then this is good to go

**Pull request merged**